File: mixin-base-test/src/test/java/net/yakclient/apitests/RegistryTest.java
package net.yakclient.apitests;


import net.questcraft.apitests.MixinTestCase;
import net.yakclient.mixin.base.registry.MixinRegistry;
import net.yakclient.mixin.base.registry.RegistryConfigurator;
import org.junit.jupiter.api.Test;

import java.lang.reflect.Constructor;
import java.lang.reflect.InvocationTargetException;

public class RegistryTest {
    public static void main(String[] args) throws ClassNotFoundException, NoSuchMethodException, IllegalAccessException, InvocationTargetException, InstantiationException {


        final RegistryConfigurator configure = RegistryConfigurator.configure().addSafePackage("net.questcraft").addTarget("net.questcraft");
        final MixinRegistry mixinRegistry = configure.create();
        {
            mixinRegistry.registerMixin(MixinTestCase.class).dumpAll();

            final Class<?> aClass = mixinRegistry.retrieveClass(net.questcraft.apitests.MixinSourceClassTest.class.getName()); //target.retrieveClass(MixinSourceClassTest.class.getName());
//            final Constructor<?> constructor = aClass.getConstructor(String.class);
//            final Object obj = constructor.newInstance("YAY");
//            obj.getClass().getMethod("printTheString", int.class).invoke(obj,11);
//            new Label().toString();
        }

        {
            mixinRegistry.registerMixin(net.questcraft.apitests.SecondMixinTestCase.class).dumpAll();


            final Class<?> aClass = mixinRegistry.retrieveClass(net.questcraft.apitests.MixinSourceClassTest.class.getName());

            final Constructor<?> constructor = aClass.getConstructor(String.class);
            final Object obj = constructor.newInstance("YAY");
            aClass.getMethod("printTheString").invoke(obj);
        }

    }

    @Test
    public void testMixinRegistry() throws ClassNotFoundException {
        System.out.println("GOt here");
        RegistryConfigurator.configure().create().registerMixin(MixinTestCase.class).dumpAll();
        printSomething();
    }


    public void printSomething() {
        System.out.println("Something");
    }
}

File: mixin-base-test/src/test/kotlin/net/yakclient/apitests/RegistryTest.kt
package net.yakclient.apitests

import net.questcraft.apitests.MixinSourceClassTest
import net.questcraft.apitests.MixinTestCase
import net.yakclient.mixin.base.registry.MixinRegistry
import net.yakclient.mixin.base.target.PackageTarget
import org.junit.jupiter.api.Test
import java.io.File

class RegistryTest {
    @Test
    fun testRegistryMixins() {
        val mixinRegistry = MixinRegistry()
        val context = mixinRegistry.applyTarget(PackageTarget.of("net.questcraft"))
        val lib = mixinRegistry.registerLib(
            File("${File(System.getProperty("user.dir")).parent}/lib/guava-17.0-TESTJAR.jar").toURI().toURL()
        )

        mixinRegistry.registerMixin(net.questcraft.apitests.SecondMixinTestCase::class.java).registerMixin(MixinTestCase::class.java).registerAll()

        val aClass = context.findClass(MixinSourceClassTest::class.java.name)
        val constructor = aClass.getConstructor(String::class.java)
        val obj = constructor.newInstance("YAY")
        aClass.getMethod("printTheString").invoke(obj)

        println(lib.findClass("com.google.common.math.DoubleMath").getMethod("isPowerOfTwo", Double::class.java).invoke(null, 5.0))
    }

}